src/DocChecks.jl

```diff
@@ -366,6 +378,43 @@ function print_indented(buffer::IO, str::AbstractString; indent = 4)
     end
 end
 
+function fix_doctest(result::Result, str, doc::Documents.Document)
+    code = result.block.code
+    filename = Base.find_source_file(result.file)
+    # read the file containing the code block
+    content = open(f -> read(f, String), filename)
+    # output stream
+    io = Compat.IOBuffer(sizehint = sizeof(content))
+    # first look for the entire code block
+    codeidx = Compat.findnext(code, content, 1)
+    if codeidx === nothing
+        Utilities.warn("Could not find code block in source file")
+        return
+    end
+    # write everything up until the code block
+    write(io, content[1:prevind(content, first(codeidx))])
```

Comment:
I was able to trigger a

```
ERROR: LoadError: BoundsError: attempt to access "# Test\r\n\r\n```jldoctest\r\njulia> 2 + 2\r\n0000\r\n```\r\n"
  at index [0]
Stacktrace:
 [1] prevind(::String, ::Int64, ::Int64) at ./strings/basic.jl:426
 [2] prevind at ./strings/basic.jl:421 [inlined]
 [3] fix_doctest(::Documenter.DocChecks.Result, ::SubString{String}, ::Documenter.Documents.Document) at /home/morten/.julia/v0.7/Documenter/src/DocChecks.jl:395
 [4] checkresult(::Module, ::Documenter.DocChecks.Result, ::Dict{Symbol,Any}, ::Documenter.Documents.Document) at /home/morten/.julia/v0.7/Documenter/src/DocChecks.jl:305
```

with [CRLFs (this file)](https://gist.githubusercontent.com/mortenpi/8ad8460373222d24869ed7eb9cbd86fa/raw/86a595fcbac03ab4c960352bc0333688094c444a/index.md) right now. So this manages to avoid your checks. Sorry for keeping on breaking this :laughing:

Reply:
Hmm.. How? I simply get:
```
 !! Could not find code block in source file
```

Reply:
Could it be that you don't have https://github.com/JuliaLang/Compat.jl/commit/5ff98924e8f010c9cf9f83758655a7f8025dbeb3 ?

Reply:
Ah, I think you're right, my bad. I happened to be running this with my 0.7 binary and I think it is a bit too old (9 days).